docs/classes.rst

```diff
@@ -210,6 +211,59 @@ There are already some utility classes implemented in ``pytest-mh``. See
     Each change that is made through the utility object (such as writing to a
     file) is automatically reverted (the original file is restored).
 
+TopologyController
+==================
+
+Topology controller can be assigned to a topology via `@pytest.mark.topology`
+or through known topology class. This controller provides various methods to
+control the topology behavior:
+
+* per-topology setup and teardown, called once before the first test/after the
+  last test for given topology is executed
+* per-test topology setup and teardown, called before and after every test case
+  for given topology
+* check topology requirements and skip the test if these are not satisfied
+
+In order to use the controller, you need to inherit from
+:class:`~pytest_mh.TopologyController` and override desired methods. Each method
+can take any parameter as defined by the topology fixtures. The parameter value
+is an instance of a :class:`~pytest_mh.MultihostHost` object.
+
+See :class:`~pytest_mh.TopologyController` for API documentation
+
+.. code-block: python
+    :caption: Topology controller example
+
+    class ExampleController(TopologyController):
+        # Override methods that you wish to use
+
+        def skip(self, client: ClientHost) -> str | None:
+            result = client.ssh.run(check_requirements, raise_on_error=False)
+            if result.rc != 0:
+                return "Topology requirements were not met"
+
+            return None
+
+    # Set the controller using the pytest marker
+    @pytest.mark.topology(
+        "example", Topology(TopologyDomain("example", client=1)),
```

Comment:
Why not the simplified `@pytest.mark.topology(KnownTopology.EXAMPLE)` here?

Reply:
and instead use the simplified `@pytest.mark.topology(KnownTopology.EXAMPLE)` here